### PR TITLE
[Internal][Executor] Move resolve data and input mapping logic to batch inputs processor

### PR DESCRIPTION
**Pull request by PeiwenGaoMS**
# Description

Move resolve data and input mapping logic to batch inputs processor.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/executor/flow_executor.py [hunk 1]
```diff
@@ -1000,202 +965,10 @@ def apply_inputs_mapping(
...
-            raise InputMappingError(
-                message_format=(
-                    "The input for batch run is incorrect. Could not find one complete line on the provided input. "
-                    "Please ensure that you supply data on the same line to resolve this issue."
-                )
-            )
-
-        result = [FlowExecutor.apply_inputs_mapping(item, inputs_mapping) for item in merged_list]
-        return result
+        return BatchInputsProcessor.apply_inputs_mapping(inputs, inputs_mapping)
```
Comment (guming-llc) [inline src/promptflow/promptflow/executor/flow_executor.py:971]: Put this to a common util function, and import it in FlowExecutor, BatchInputsProcessor and promptflow._internal?

Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/executor/flow_executor.py]: Updated.

File: src/promptflow/promptflow/batch/_batch_inputs_processor.py [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+from promptflow.batch._errors import InputMappingError
+from promptflow.contracts.flow import FlowInputDefinition
+from promptflow.executor.flow_executor import LINE_NUMBER_KEY
+
+
+class BatchInputsProcessor:
+    def __init__(
+        self,
+        working_dir: Path,
+        flow_inputs: Dict[str, FlowInputDefinition],
```
Comment (thy09) [inline src/promptflow/promptflow/batch/_batch_inputs_processor.py:22]: This type should be Mapping instead of Dict? since we actually should not edit it.

Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/batch/_batch_inputs_processor.py:22]: Updated.

File: src/promptflow/promptflow/batch/_batch_inputs_processor.py [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+    :param inputs: A mapping of input keys to their corresponding values.
+    :type inputs: Mapping[str, Mapping[str, Any]]
+    :param inputs_mapping: A mapping of input keys to their corresponding mapping expressions.
+    :type inputs_mapping: Mapping[str, str]
+    :return: A dictionary of input keys to their corresponding mapped values.
+    :rtype: Dict[str, Any]
+    :raises InputMappingError: If any of the input mapping relations are not found in the inputs.
+    """
+    import re
```
Comment (guming-llc) [inline src/promptflow/promptflow/batch/_batch_inputs_processor.py:252]: Why is this not in file top?

Comment (PeiwenGaoMS) [reply src/promptflow/promptflow/batch/_batch_inputs_processor.py:252]: Just copy from `flow_executor`. I'll update it.

**Review by guming-llc — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**